packages/shared/src/hooks/search/useSearchResultsLayout.ts

```diff
@@ -0,0 +1,29 @@
+import { useViewSize, ViewSize } from '../useViewSize';
+import { useConditionalFeature } from '../useConditionalFeature';
+import { feature } from '../../lib/featureManagement';
+import { SharedFeedPage } from '../../components/utilities';
+import { useActiveFeedNameContext } from '../../contexts';
+
+interface SearchResultsLayout {
+  isSearchResultsUpgrade: boolean;
+}
+
+export const useSearchResultsLayout = (): SearchResultsLayout => {
+  const { feedName } = useActiveFeedNameContext();
+  const isLaptop = useViewSize(ViewSize.Laptop);
+  const { value: isSearchUpgradeExperiment, isLoading } = useConditionalFeature(
+    {
+      feature: feature.searchResultsUpgrade,
+      shouldEvaluate: !!isLaptop && feedName === SharedFeedPage.Search,
```

Comment:
We should put this condition in a variable and use it on the condition for `isSearchResultsUpgrade`. The thing is, once the evaluation has finished, the result stays the same even if the value of the `shouldEvaluate` has changed.

For example: if you are viewing the search page, the result of `isSearchUpgradeExperiment` then becomes true, which would make `isSearchResultsUpgrade` become `true` also (let's say the other conditions are true as well). Now, if you switch pages, even though the `shouldEvaluate` has changed, the value of `isSearchUpgradeExperiment` from the cache is still the same which is still `true`.